bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/BoschHttpClient.java

```diff
@@ -0,0 +1,139 @@
+package org.openhab.binding.boschshc.internal;
+
+import static org.eclipse.jetty.http.HttpMethod.GET;
+
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStoreException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+public class BoschHttpClient extends HttpClient {
+
+    private static final int MAX_PAIR_TRIES = 12;
```

Comment:
I'm assuming the pairing gets triggered during binding initialization? Wondering if 12 is insufficient and people will end up clicking around in the UI forever and it will never connect because they don't get that they have to restart openhab?

Reply:
We can tweek the values however we want. The 12 retries together with a sleep of 15s gives users a time of 15s * 12 = 180s = 3 minutes for paring.
I will take a look into the Hue Binding learn from other examples.